src/vehicle/router.cpp

```diff
@@ -18,10 +18,13 @@ namespace CityFlow {
         iCurRoad = route.begin();
     }
 
-    Router::Router(Vehicle *vehicle, std::shared_ptr<const Route> route, std::mt19937 *rnd)
+    Router::Router(Vehicle *vehicle, std::shared_ptr<const Route> route, std::mt19937 *rnd, bool update)
```

Comment:
can you explain this change?

Reply:
When we  create a Router object, sometimes we need to update the route according to the anchor points (when pushing a new vehicle to a lane, we need to recompute the path with the least expected traveling time), and sometimes we don't (when read from the archive file where the whole route is exactly saved)